### DISCUSSION TITLE
Suggestion that might reduce startup time

### DISCUSSION THREAD

**Discussion by yazdan** [DNSCrypt/dnscrypt-proxy#1666] [Ideas]
Hi,

First thanks for all your effort making this happen.

I run my dnscrypt-proxy on NanoPi Neo2 which is a little limited in resources. Calling `proxy.serversInfo.refresh(proxy)` in `StartProxy` in `proxy.go` takes around 3 mins on NanoPiNeo2 and I think it can be reduced.

What I'm suggesting is:

- Move `proxy.serversInfo.refresh(proxy)` to a go routine.
- Wait until at least one server is available
- Start to serve requests
- Wait for refresh to end

Please let me know If this is can be done or not? If you think this will improve the dnscrypt-proxy, I will be more than happy to try to implement it.

**Comment by jedisct1** [comment]
Hi!

The client listeners start immediately.

And as soon as at least one server is live, these servers are used and responses are sent, even if the full list of servers hasn't been processed yet.

So, calling `refresh()` in a goroutine isn't going to do anything. The listeners themselves are already running as goroutines, and can even start responding queries before the network is configured (for captive portals tests).

**Reply by yazdan** [reply]
Hi,
Thanks for your response, but it is not what I see in my setup.
Let me investigate more an I'll let you know.
